### PR TITLE
email reader

### PR DESCRIPTION
**Pull request by kashifch**
read email

### CODE REVIEW

File: cypress/integration/admin_portal/tests/read_email_template.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import HelperFunctions from '../helpers/helper_functions'
+
+describe('Email assertion:', () => {
+  const helpers = new HelperFunctions()
+  it('Gets the email on basis of a search query', function () {
+    cy.log(helpers.getUniqueEmailAddress())
```
Comment (imhassantariq) [inline cypress/integration/admin_portal/tests/read_email_template.js]: Why log? what is the purpose of the log here?

File: cypress/integration/admin_portal/tests/read_email_template.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+describe('Email assertion:', () => {
+  const helpers = new HelperFunctions()
+  it('Gets the email on basis of a search query', function () {
+    const mailOptions = {
+      from: 'activate@edx.org',
+      to: helpers.getUniqueEmailAlias(),
+      subject: 'Activate',
+    }
+    cy.task('mailReader', mailOptions).then((email) => {
+      cy.log(helpers.extractUrlFromEmail(email, 'activate'))
```
Comment (muhammad-ammar) [inline cypress/integration/admin_portal/tests/read_email_template.js]: Don't we need to assert/verify anything in this test?

File: cypress/integration/admin_portal/tests/read_email_template.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import HelperFunctions from '../helpers/helper_functions'
+
+describe('Email assertion:', () => {
+  const helpers = new HelperFunctions()
```
[the remark this answers is not in the document]
Comment (kashifch) [reply cypress/integration/admin_portal/tests/read_email_template.js]: done

File: cypress/integration/admin_portal/helpers/helper_functions.js [hunk 4]
```diff
@@ -1,11 +1,40 @@
...
+    const position = emailAddress.indexOf('@')
+    return `${emailAddress.slice(0, position)}+${randomString}${emailAddress.slice(position)}`
+  }
+
+  /**
+ * Helper method to extract url from email text
+ *
+ * @param {String} emailText Email body from which url needs to be searched
+ * @param {String} subString any substring which should ne present in the target url
+ */
```
[the remark this answers is not in the document]
Comment (kashifch) [reply cypress/integration/admin_portal/helpers/helper_functions.js]: done

File: cypress/integration/admin_portal/helpers/helper_functions.js [hunk 5]
```diff
@@ -1,11 +1,40 @@
...
   verifyLinksAndText(elems, expectedDict) {
     const names = [...elems].map(el => el.textContent.trim())
     const urls = [...elems].map(el => el.getAttribute('href'))
     const actualDict = Object.assign({}, ...names.map((n, index) => ({ [n]: urls[index] })))
     expect(actualDict).to.deep.equal(expectedDict)
   }
+
+  /**
+ * Helper method to create unique gmail alias
+ */
```
[the remark this answers is not in the document]
Comment (kashifch) [reply cypress/integration/admin_portal/helpers/helper_functions.js]: done

File: cypress/integration/admin_portal/tests/read_email_template.js (hunk 2 above)
Comment (kashifch) [reply cypress/integration/admin_portal/tests/read_email_template.js]: it's just a template to show email reading is working, actual tests will be added afterwards

File: cypress/plugins/gmail_reader.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
+const { google } = require('googleapis')
```
[the remark this answers is not in the document]
Comment (kashifch) [reply cypress/plugins/gmail_reader.js]: yes thats the main file I want you to review as it's not cypress and kind of my first attempt to write javascript.

**Pull request merged**